<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>R Notebook</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Code</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tytell/BaxterVertEvol">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">R Notebook</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Code/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211230code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20211230)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211230code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211230)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtytellBaxterVertEvoltreebbb4c584487b865ad3c5b90b1c6482c1a14fae14targetblankbbb4c58a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tytell/BaxterVertEvol/tree/bbb4c584487b865ad3c5b90b1c6482c1a14fae14" target="_blank">bbb4c58</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtytellBaxterVertEvoltreebbb4c584487b865ad3c5b90b1c6482c1a14fae14targetblankbbb4c58a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tytell/BaxterVertEvol/tree/bbb4c584487b865ad3c5b90b1c6482c1a14fae14" target="_blank">bbb4c58</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    Flexibility Comparisons.nb.html
    Ignored:    Main.nb.html
    Ignored:    PGLS.FullData.nb.html
    Ignored:    PGLSforeachMeasFeature.nb.html
    Ignored:    PGLSwithPCA_Dims.nb.html
    Ignored:    PreppedVertMeas.nb.html
    Ignored:    ProcessCymatogasterFiles.nb.html
    Ignored:    ProcessFCSVfiles.nb.html
    Ignored:    TestingHabitatwithFriedmanData.nb.html
    Ignored:    Trilok_tree.nb.html
    Ignored:    VertLM.nb.html
    Ignored:    VertMeasLDA_Attempt.nb.html
    Ignored:    VertPGLS.nb.html
    Ignored:    VertPairs.nb.html
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/02-CheckSpeciesMatch.nb.html
    Ignored:    analysis/10-VertLM.nb.html
    Ignored:    analysis/12-VertPGLS2.nb.html
    Ignored:    analysis/13-VertPGLS-MANOVA40.nb.html
    Ignored:    analysis/13-VertPGLS-MANOVA90.nb.html
    Ignored:    analysis/14-VertPGLS-MANOVA40.nb.html
    Ignored:    analysis/20-plot_phylogeny.nb.html
    Ignored:    analysis/21-plot_fits_and_summary.nb.html
    Ignored:    analysis/CheckSpeciesMatch.nb.html
    Ignored:    caper_test.nb.html
    Ignored:    data/.DS_Store
    Ignored:    ggtree_attempt.nb.html
    Ignored:    output/.DS_Store
    Ignored:    plot_example_data.nb.html
    Ignored:    plot_fits_and_summary.nb.html
    Ignored:    plot_phylogeny.nb.html
    Ignored:    summarize_vert_meas.nb.html
    Ignored:    test_phylogeny.nb.html
    Ignored:    test_vertebraspace.nb.html
    Ignored:    vert_evol.Rproj

Untracked files:
    Untracked:  Archive.zip
    Untracked:  Main.html
    Untracked:  ProcessFCSVfiles.Rmd
    Untracked:  VertPGLS.html
    Untracked:  analysis/13-VertPGLS-MANOVA90.Rmd
    Untracked:  analysis/14-VertPGLS-MANOVA40.Rmd
    Untracked:  data/actinopt_12k_raxml.tre.xz
    Untracked:  gg_saver.R
    Untracked:  output/BodyDistribution.pdf
    Untracked:  output/MasterVert_Measurements.csv
    Untracked:  output/anova_table.rtf
    Untracked:  output/anovatabs.csv
    Untracked:  output/effectsizes.csv
    Untracked:  output/fineness.pdf
    Untracked:  output/habitatvals.csv
    Untracked:  output/manova_table.rtf
    Untracked:  output/mean_d_alphaPos_CBL.pdf
    Untracked:  output/mean_params.pdf
    Untracked:  output/pair_plot.pdf
    Untracked:  output/phylogeny_families.pdf
    Untracked:  output/plot_example_data_figure.pdf
    Untracked:  output/predvals.Rds
    Untracked:  output/stats_table.rtf
    Untracked:  output/vertdata_show_species.csv
    Untracked:  output/vertdata_summary_species.csv
    Untracked:  output/vertshape.pdf
    Untracked:  plot_fits_and_summary.Rmd
    Untracked:  summarize_vert_meas.html
    Untracked:  testtree.csv
    Untracked:  vert_tree.csv

Unstaged changes:
    Deleted:    .Rprofile
    Modified:   analysis/index.Rmd
    Modified:   data/MasterVert_Measurements.csv
    Modified:   output/MasterVert_Measurements_Matched.csv
    Modified:   output/vert_tree.rds
    Modified:   output/vertdata_centered.csv
    Modified:   output/vertdata_summary.csv
    Deleted:    renv.lock
    Deleted:    renv/.gitignore
    Deleted:    renv/activate.R
    Deleted:    renv/settings.dcf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/11-MergePhylogeny.Rmd</code>) and
HTML (<code>docs/11-MergePhylogeny.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tytell/BaxterVertEvol/blob/bbb4c584487b865ad3c5b90b1c6482c1a14fae14/analysis/11-MergePhylogeny.Rmd" target="_blank">bbb4c58</a>
</td>
<td>
Eric Tytell
</td>
<td>
2022-10-23
</td>
<td>
Final analysis scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tytell/BaxterVertEvol/blob/65e217eb4dfde937a04837c1d98cc6d615b8f3f5/analysis/11-MergePhylogeny.Rmd" target="_blank">65e217e</a>
</td>
<td>
Eric Tytell
</td>
<td>
2022-09-16
</td>
<td>
Updated code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tytell/BaxterVertEvol/blob/23b4a3f77ea415ec1bfc2d4377816eecae66a86a/analysis/11-MergePhylogeny.Rmd" target="_blank">23b4a3f</a>
</td>
<td>
Eric Tytell
</td>
<td>
2022-09-06
</td>
<td>
Starting to compare a different phylogeny
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2 ──
✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.8     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.1
✔ readr   2.1.2     ✔ forcats 0.5.2
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(ggbeeswarm)
library(phytools)</code></pre>
<pre><code>Loading required package: ape
Loading required package: maps

Attaching package: &#39;maps&#39;

The following object is masked from &#39;package:purrr&#39;:

    map</code></pre>
<pre class="r"><code>library(patchwork)
library(here)</code></pre>
<pre><code>here() starts at /Users/etytel01/Documents/Vertebrae/Code</code></pre>
<pre class="r"><code>library(nlme)</code></pre>
<pre><code>
Attaching package: &#39;nlme&#39;

The following object is masked from &#39;package:dplyr&#39;:

    collapse</code></pre>
<pre class="r"><code>library(ape)
library(geiger)
library(ggtree)</code></pre>
<pre><code>Warning: package &#39;ggtree&#39; was built under R version 4.2.1</code></pre>
<pre><code>ggtree v3.4.2 For help: https://yulab-smu.top/treedata-book/

If you use the ggtree package suite in published research, please cite
the appropriate paper(s):

Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam.
ggtree: an R package for visualization and annotation of phylogenetic
trees with their covariates and other associated data. Methods in
Ecology and Evolution. 2017, 8(1):28-36. doi:10.1111/2041-210X.12628

G Yu. Data Integration, Manipulation and Visualization of Phylogenetic
Trees (1st ed.). Chapman and Hall/CRC. 2022. ISBN: 9781032233574

LG Wang, TTY Lam, S Xu, Z Dai, L Zhou, T Feng, P Guo, CW Dunn, BR
Jones, T Bradley, H Zhu, Y Guan, Y Jiang, G Yu. treeio: an R package
for phylogenetic tree input and output with richly annotated and
associated data. Molecular Biology and Evolution. 2020, 37(2):599-603.
doi: 10.1093/molbev/msz240

Attaching package: &#39;ggtree&#39;

The following object is masked from &#39;package:nlme&#39;:

    collapse

The following object is masked from &#39;package:ape&#39;:

    rotate

The following object is masked from &#39;package:tidyr&#39;:

    expand</code></pre>
<pre class="r"><code>library(emmeans)
library(car)</code></pre>
<pre><code>Loading required package: carData

Attaching package: &#39;car&#39;

The following object is masked from &#39;package:dplyr&#39;:

    recode

The following object is masked from &#39;package:purrr&#39;:

    some</code></pre>
<pre class="r"><code>library(Hmisc)</code></pre>
<pre><code>Loading required package: lattice
Loading required package: survival
Loading required package: Formula

Attaching package: &#39;Hmisc&#39;

The following object is masked from &#39;package:ape&#39;:

    zoom

The following objects are masked from &#39;package:dplyr&#39;:

    src, summarize

The following objects are masked from &#39;package:base&#39;:

    format.pval, units</code></pre>
<pre class="r"><code>citation()</code></pre>
<pre><code>
To cite R in publications use:

  R Core Team (2022). R: A language and environment for statistical
  computing. R Foundation for Statistical Computing, Vienna, Austria.
  URL https://www.R-project.org/.

A BibTeX entry for LaTeX users is

  @Manual{,
    title = {R: A Language and Environment for Statistical Computing},
    author = {{R Core Team}},
    organization = {R Foundation for Statistical Computing},
    address = {Vienna, Austria},
    year = {2022},
    url = {https://www.R-project.org/},
  }

We have invested a lot of time and effort in creating R, please cite it
when using it for data analysis. See also &#39;citation(&quot;pkgname&quot;)&#39; for
citing R packages.</code></pre>
<pre class="r"><code>print(getRversion())</code></pre>
<pre><code>[1] &#39;4.2.0&#39;</code></pre>
<pre class="r"><code>citation(&quot;nlme&quot;)</code></pre>
<pre><code>
To cite package &#39;nlme&#39; in publications use:

  Pinheiro J, Bates D, R Core Team (2022). _nlme: Linear and Nonlinear
  Mixed Effects Models_. R package version 3.1-159,
  &lt;https://CRAN.R-project.org/package=nlme&gt;.

  Pinheiro JC, Bates DM (2000). _Mixed-Effects Models in S and S-PLUS_.
  Springer, New York. doi:10.1007/b98882
  &lt;https://doi.org/10.1007/b98882&gt;.

To see these entries in BibTeX format, use &#39;print(&lt;citation&gt;,
bibtex=TRUE)&#39;, &#39;toBibtex(.)&#39;, or set
&#39;options(citation.bibtex.max=999)&#39;.</code></pre>
<pre class="r"><code>packageVersion(&quot;nlme&quot;)</code></pre>
<pre><code>[1] &#39;3.1.159&#39;</code></pre>
<pre class="r"><code>citation(&quot;ape&quot;)</code></pre>
<pre><code>
To cite ape in a publication please use:

  Paradis E. &amp; Schliep K. 2019. ape 5.0: an environment for modern
  phylogenetics and evolutionary analyses in R. Bioinformatics 35:
  526-528.

A BibTeX entry for LaTeX users is

  @Article{,
    title = {ape 5.0: an environment for modern phylogenetics and evolutionary analyses in {R}},
    author = {E. Paradis and K. Schliep},
    journal = {Bioinformatics},
    year = {2019},
    volume = {35},
    pages = {526-528},
  }

As ape is evolving quickly, you may want to cite also its version
number (found with &#39;library(help = ape)&#39; or &#39;packageVersion(&quot;ape&quot;)&#39;).</code></pre>
<pre class="r"><code>packageVersion(&quot;ape&quot;)</code></pre>
<pre><code>[1] &#39;5.6.2&#39;</code></pre>
<pre class="r"><code>citation(&quot;geiger&quot;)</code></pre>
<pre><code>
To cite medusa, auteur, or geiger in a publication use:

medusa

  Alfaro Michael E, Francesco Santini, Chad Brock, Hugo Alamillo, Alex
  Dornburg, Daniel L Rabosky, Giorgio Carnevale, and Luke J Harmon.
  2009. Nine exceptional radiations plus high turnover explain species
  diversity in jawed vertebrates. PNAS 106:13410-13414.

auteur

  Eastman Jonathan M, Michael E Alfaro, Paul Joyce, Andrew L Hipp, and
  Luke J Harmon. 2011. A novel comparative method for identifying
  shifts in the rate of character evolution on trees. Evolution
  65:3578-3589.

MECCA

  Slater Graham J, Luke J Harmon, Daniel Wegmann, Paul Joyce, Liam J
  Revell, and Michael E Alfaro. 2012. Fitting models of continuous
  trait evolution to incompletely sampled comparative data using
  approximate Bayesian computation. Evolution 66:752-762.

geiger-orig

  Harmon Luke J, Jason T Weir, Chad D Brock, Richard E Glor, and
  Wendell Challenger. 2008. GEIGER: investigating evolutionary
  radiations. Bioinformatics 24:129-131.

geiger

  Pennell Matthew W, Jonathan M Eastman, Graham J Slater, Joseph W
  Brown, Josef C Uyeda, Richard G FitzJohn, Michael E Alfaro, and Luke
  J Harmon. 2014. geiger v2.0: an expanded suite of methods for fitting
  macroevolutionary models to phylogenetic trees. Bioinformatics
  30:2216-2218.

To see these entries in BibTeX format, use &#39;print(&lt;citation&gt;,
bibtex=TRUE)&#39;, &#39;toBibtex(.)&#39;, or set
&#39;options(citation.bibtex.max=999)&#39;.</code></pre>
<pre class="r"><code>packageVersion(&quot;geiger&quot;)</code></pre>
<pre><code>[1] &#39;2.0.10&#39;</code></pre>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<pre class="r"><code>vertdata_sum &lt;- read_csv(here(&quot;output/vertdata_summary.csv&quot;)) |&gt; 
  mutate(MatchSpecies = str_c(MatchGenus, MatchSpecies, sep = &#39;_&#39;)) |&gt; 
  select(-MatchGenus)</code></pre>
<pre><code>Rows: 82 Columns: 46
── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr  (8): Species, Habitat, Water_Type, MatchSpecies, MatchGenus, alltaxon, ...
dbl (38): Indiv, fineness, CBL_med, CBL_max, CBL_mn, d_med, d_max, d_mn, alp...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<div id="phylogeny" class="section level2">
<h2>Phylogeny</h2>
<p>This is the tree from the Fish Tree of Life: <a
href="https://fishtreeoflife.org/"
class="uri">https://fishtreeoflife.org/</a></p>
<pre class="r"><code>tree &lt;- read.tree(here(&#39;data/actinopt_12k_treePL.tre.xz&#39;))</code></pre>
<p>Get the names of species from the tree.</p>
<pre class="r"><code>allspecies &lt;- tibble(tree$tip.label)
colnames(allspecies) &lt;- c(&#39;Species&#39;)
head(allspecies)</code></pre>
<pre><code># A tibble: 6 × 1
  Species          
  &lt;chr&gt;            
1 Gambusia_marshi  
2 Gambusia_panuco  
3 Gambusia_regani  
4 Gambusia_aurata  
5 Gambusia_hurtadoi
6 Gambusia_gaigei  </code></pre>
<p>Set up the tip number (just the row)</p>
<pre class="r"><code>allspecies$Tip &lt;- seq_len(nrow(allspecies))</code></pre>
<p>The double species name for Fundulus confuses things, so we’ll just
correct it manually.</p>
<pre class="r"><code>vertdata_sum &lt;-
  vertdata_sum |&gt; 
  mutate(MatchSpecies = if_else(Species == &#39;Fundulus_heteroclitus&#39;,
                                &#39;Fundulus_heteroclitus_heteroclitus&#39;,
                                MatchSpecies))</code></pre>
<pre class="r"><code>vertdata &lt;- left_join(vertdata_sum, allspecies, 
                                  by=c(&quot;MatchSpecies&quot;=&quot;Species&quot;)) %&gt;%
  select(-(ends_with(&quot;.x&quot;) | ends_with(&quot;.y&quot;)))
vertdata</code></pre>
<pre><code># A tibble: 82 × 46
   Species    Indiv Habitat Water…¹ Match…² allta…³ Order Family finen…⁴ CBL_med
   &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt;   &lt;dbl&gt;
 1 Abramis_b…     1 pelagic freshw… Alburn… Actino… Cypr… Cypri…    8.95 0.0166 
 2 Alectis_c…     1 pelagic marine  Alecti… Actino… Cara… Caran…    8.75 0.0346 
 3 Alosa_pse…     1 pelagic anadro… Alosa_… Actino… Clup… Clupe…    7.39 0.0165 
 4 Amia_calva     1 demers… freshw… Amia_c… Actino… Amii… Amiid…    6.72 0.00983
 5 Ammodytes…     1 demers… marine  Ammody… Actino… Uran… Ammod…   16.9  0.0132 
 6 Anodontos…     1 pelagic freshw… Doroso… Actino… Clup… Clupe…    4.66 0.0228 
 7 Anoplarch…     1 benthic marine  Anopla… Actino… Perc… Stich…    8.62 0.0195 
 8 Anoplarch…     1 demers… marine  Anopla… Actino… Perc… Stich…    9.51 0.0156 
 9 Anoplogas…     1 pelagic marine  Anoplo… Actino… Bery… Anopl…    5.04 0.0287 
10 Aphareus_…     1 pelagic marine  Aphare… Actino… Ince… Lutja…    5.01 0.0312 
# … with 72 more rows, 36 more variables: CBL_max &lt;dbl&gt;, CBL_mn &lt;dbl&gt;,
#   d_med &lt;dbl&gt;, d_max &lt;dbl&gt;, d_mn &lt;dbl&gt;, alphaAnt_med &lt;dbl&gt;,
#   alphaAnt_max &lt;dbl&gt;, alphaAnt_mn &lt;dbl&gt;, alphaPos_med &lt;dbl&gt;,
#   alphaPos_max &lt;dbl&gt;, alphaPos_mn &lt;dbl&gt;, DAnt_med &lt;dbl&gt;, DAnt_max &lt;dbl&gt;,
#   DAnt_mn &lt;dbl&gt;, DPos_med &lt;dbl&gt;, DPos_max &lt;dbl&gt;, DPos_mn &lt;dbl&gt;,
#   dBW_med &lt;dbl&gt;, dBW_max &lt;dbl&gt;, dBW_mn &lt;dbl&gt;, DAntBW_med &lt;dbl&gt;,
#   DAntBW_max &lt;dbl&gt;, DAntBW_mn &lt;dbl&gt;, DPosBW_med &lt;dbl&gt;, DPosBW_max &lt;dbl&gt;, …</code></pre>
<p>Drop species without a match</p>
<pre class="r"><code>vertdata %&gt;%
  filter(is.na(Tip)) %&gt;%
  distinct(Species, .keep_all=TRUE) %&gt;%
  select(Species, MatchSpecies, Tip, Habitat)</code></pre>
<pre><code># A tibble: 0 × 4
# … with 4 variables: Species &lt;chr&gt;, MatchSpecies &lt;chr&gt;, Tip &lt;int&gt;,
#   Habitat &lt;chr&gt;</code></pre>
<pre class="r"><code>vertdata &lt;-
  vertdata %&gt;%
  filter(!is.na(Tip))</code></pre>
<pre class="r"><code>ourspecies &lt;-
  vertdata %&gt;%
  distinct(Species, .keep_all=TRUE)</code></pre>
<pre class="r"><code>verttree &lt;- keep.tip(tree, tip=as.vector(ourspecies$Tip))</code></pre>
<p>Replace the names of species in the tree with our matched ones.</p>
<pre class="r"><code>verttree$tip.label &lt;-
  ourspecies |&gt; 
  arrange(Tip) |&gt; 
  pull(Species)</code></pre>
<pre class="r"><code>plotTree(verttree)</code></pre>
<p><img src="figure/11-MergePhylogeny.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>vertdata_sp &lt;- 
  vertdata %&gt;%
  distinct(Species, .keep_all = TRUE) %&gt;%
  mutate(rowname = Species) %&gt;%
  column_to_rownames(var = &quot;rowname&quot;)</code></pre>
<pre class="r"><code>left_join(as_tibble(verttree),
          vertdata_sp %&gt;%
            rownames_to_column(&quot;label&quot;) %&gt;%
            select(label, Habitat)) %&gt;%
  tidytree::as.treedata() %&gt;%
  ggtree(layout = &#39;circular&#39;) + # geom_tiplab() +
  geom_tippoint(aes(color = Habitat))</code></pre>
<pre><code>Joining, by = &quot;label&quot;</code></pre>
<p><img src="figure/11-MergePhylogeny.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" />
Check if tree and data match</p>
<pre class="r"><code>length(verttree$tip.label)</code></pre>
<pre><code>[1] 82</code></pre>
<pre class="r"><code>nrow(vertdata_sp)</code></pre>
<pre><code>[1] 82</code></pre>
<pre class="r"><code>name.check(verttree, vertdata_sp)</code></pre>
<pre><code>[1] &quot;OK&quot;</code></pre>
<p>Merge the measurements and the tree</p>
<pre class="r"><code>verttree_data &lt;-
  as_tibble(verttree) %&gt;%
  left_join(vertdata_sp %&gt;%
              rownames_to_column(&quot;label&quot;) %&gt;%
              select(label, Habitat, alltaxon, Order, Family, fineness,
                     ends_with(&quot;80&quot;), ends_with(&quot;max&quot;), ends_with(&quot;med&quot;), ends_with(&quot;slope&quot;), ends_with(&quot;quad&quot;)))</code></pre>
<pre><code>Joining, by = &quot;label&quot;</code></pre>
<p>Save out the tree data. There is something subtly different about
saving the data out as a csv file and saving it as an RDS file. The base
class of the tree is tbl_tree, but when we load it back in from a csv,
despite having exactly the same data, it won’t work with the tidytree
functions. So we save it in an RDS file, which preserves the class.</p>
<pre class="r"><code>write_csv(vertdata_sp, here(&#39;output/vertdata_summary_species.csv&#39;))
saveRDS(verttree, here(&#39;output/vert_tree.rds&#39;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] Hmisc_4.7-1      Formula_1.2-4    survival_3.3-1   lattice_0.20-45 
 [5] car_3.1-0        carData_3.0-5    emmeans_1.8.0    ggtree_3.4.2    
 [9] geiger_2.0.10    nlme_3.1-159     here_1.0.1       patchwork_1.1.2 
[13] phytools_1.2-0   maps_3.4.0       ape_5.6-2        ggbeeswarm_0.6.0
[17] forcats_0.5.2    stringr_1.4.1    dplyr_1.0.9      purrr_0.3.4     
[21] readr_2.1.2      tidyr_1.2.0      tibble_3.1.8     ggplot2_3.3.6   
[25] tidyverse_1.3.2 

loaded via a namespace (and not attached):
  [1] readxl_1.4.1            backports_1.4.1         fastmatch_1.1-3        
  [4] workflowr_1.7.0         igraph_1.3.4            lazyeval_0.2.2         
  [7] splines_4.2.0           digest_0.6.29           yulab.utils_0.0.5      
 [10] htmltools_0.5.3         fansi_1.0.3             checkmate_2.1.0        
 [13] magrittr_2.0.3          optimParallel_1.0-2     googlesheets4_1.0.1    
 [16] cluster_2.1.3           tzdb_0.3.0              modelr_0.1.9           
 [19] vroom_1.5.7             jpeg_0.1-9              colorspace_2.0-3       
 [22] rvest_1.0.3             haven_2.5.1             xfun_0.32              
 [25] crayon_1.5.1            jsonlite_1.8.0          phangorn_2.9.0         
 [28] glue_1.6.2              gtable_0.3.1            gargle_1.2.0           
 [31] abind_1.4-5             scales_1.2.1            mvtnorm_1.1-3          
 [34] DBI_1.1.3               Rcpp_1.0.9              plotrix_3.8-2          
 [37] htmlTable_2.4.1         xtable_1.8-4            gridGraphics_0.5-1     
 [40] tidytree_0.4.0          bit_4.0.4               foreign_0.8-82         
 [43] subplex_1.8             deSolve_1.33            htmlwidgets_1.5.4      
 [46] httr_1.4.4              RColorBrewer_1.1-3      ellipsis_0.3.2         
 [49] farver_2.1.1            pkgconfig_2.0.3         nnet_7.3-17            
 [52] sass_0.4.2              dbplyr_2.2.1            deldir_1.0-6           
 [55] utf8_1.2.2              labeling_0.4.2          ggplotify_0.1.0        
 [58] tidyselect_1.1.2        rlang_1.0.4             later_1.3.0            
 [61] munsell_0.5.0           cellranger_1.1.0        tools_4.2.0            
 [64] cachem_1.0.6            cli_3.3.0               generics_0.1.3         
 [67] broom_1.0.1             evaluate_0.16           fastmap_1.1.0          
 [70] yaml_2.3.5              bit64_4.0.5             knitr_1.40             
 [73] fs_1.5.2                whisker_0.4             aplot_0.1.6            
 [76] xml2_1.3.3              compiler_4.2.0          rstudioapi_0.14        
 [79] beeswarm_0.4.0          png_0.1-7               reprex_2.0.2           
 [82] treeio_1.20.2           clusterGeneration_1.3.7 bslib_0.4.0            
 [85] stringi_1.7.8           highr_0.9               Matrix_1.4-1           
 [88] vctrs_0.4.1             pillar_1.8.1            lifecycle_1.0.1        
 [91] combinat_0.0-8          jquerylib_0.1.4         estimability_1.4.1     
 [94] data.table_1.14.2       httpuv_1.6.5            R6_2.5.1               
 [97] latticeExtra_0.6-30     promises_1.2.0.1        gridExtra_2.3          
[100] vipor_0.4.5             codetools_0.2-18        MASS_7.3-56            
[103] assertthat_0.2.1        rprojroot_2.0.3         withr_2.5.0            
[106] mnormt_2.1.0            expm_0.999-6            parallel_4.2.0         
[109] hms_1.1.2               rpart_4.1.16            quadprog_1.5-8         
[112] grid_4.2.0              ggfun_0.0.7             coda_0.19-4            
[115] rmarkdown_2.16          googledrive_2.0.0       git2r_0.30.1           
[118] numDeriv_2016.8-1.1     scatterplot3d_0.3-41    lubridate_1.8.0        
[121] base64enc_0.1-3         interp_1.1-3           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
