<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Make Figure 1</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.9.4.1/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-1.57.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-1.57.1/plotly-latest.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Code</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tytell/BaxterVertEvol">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Make Figure 1</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-11
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Code/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211230code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211230)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211230code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211230)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtytellBaxterVertEvoltreee47dde2b2a99a3313f0d2dd3de04e4d203ce7af3targetblanke47dde2a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tytell/BaxterVertEvol/tree/e47dde2b2a99a3313f0d2dd3de04e4d203ce7af3" target="_blank">e47dde2</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtytellBaxterVertEvoltreee47dde2b2a99a3313f0d2dd3de04e4d203ce7af3targetblanke47dde2a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/tytell/BaxterVertEvol/tree/e47dde2b2a99a3313f0d2dd3de04e4d203ce7af3" target="_blank">e47dde2</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    Flexibility Comparisons.nb.html
    Ignored:    Main.nb.html
    Ignored:    PGLS.FullData.nb.html
    Ignored:    PGLSforeachMeasFeature.nb.html
    Ignored:    PGLSwithPCA_Dims.nb.html
    Ignored:    PreppedVertMeas.nb.html
    Ignored:    ProcessCymatogasterFiles.nb.html
    Ignored:    ProcessFCSVfiles.nb.html
    Ignored:    TestingHabitatwithFriedmanData.nb.html
    Ignored:    Trilok_tree.nb.html
    Ignored:    VertLM.nb.html
    Ignored:    VertMeasLDA_Attempt.nb.html
    Ignored:    VertPGLS.nb.html
    Ignored:    VertPairs.nb.html
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/10-VertLM.nb.html
    Ignored:    analysis/20-plot_phylogeny.nb.html
    Ignored:    analysis/21-plot_fits_and_summary.nb.html
    Ignored:    analysis/CheckSpeciesMatch.nb.html
    Ignored:    caper_test.nb.html
    Ignored:    data/.DS_Store
    Ignored:    ggtree_attempt.nb.html
    Ignored:    plot_example_data.nb.html
    Ignored:    plot_fits_and_summary.nb.html
    Ignored:    plot_phylogeny.nb.html
    Ignored:    renv/library/
    Ignored:    renv/staging/
    Ignored:    summarize_vert_meas.nb.html
    Ignored:    test_phylogeny.nb.html
    Ignored:    test_vertebraspace.nb.html
    Ignored:    vert_evol.Rproj

Untracked files:
    Untracked:  Main.html
    Untracked:  ProcessFCSVfiles.Rmd
    Untracked:  VertPGLS.html
    Untracked:  gg_saver.R
    Untracked:  output/BodyDistribution.pdf
    Untracked:  output/MasterVert_Measurements.csv
    Untracked:  output/mean_d_alphaPos_CBL.pdf
    Untracked:  output/pair_plot.pdf
    Untracked:  output/plot_example_data_figure.pdf
    Untracked:  output/stats_table.rtf
    Untracked:  plot_fits_and_summary.Rmd
    Untracked:  summarize_vert_meas.html
    Untracked:  testtree.csv
    Untracked:  vert_tree.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/20-plot_phylogeny.Rmd</code>) and HTML (<code>docs/20-plot_phylogeny.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tytell/BaxterVertEvol/blob/23908bd4e6e24d389ba137373d7d60c262968ccf/analysis/20-plot_phylogeny.Rmd" target="_blank">23908bd</a>
</td>
<td>
Eric Tytell
</td>
<td>
2021-12-30
</td>
<td>
Test site build again
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tytell/BaxterVertEvol/blob/edeae3c58db615477cae99bae2ff1e4b01b09b5d/analysis/20-plot_phylogeny.Rmd" target="_blank">edeae3c</a>
</td>
<td>
Eric Tytell
</td>
<td>
2021-12-30
</td>
<td>
Rename notebooks to indicate order
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✓ ggplot2 3.3.5     ✓ purrr   0.3.4
✓ tibble  3.1.4     ✓ dplyr   1.0.7
✓ tidyr   1.1.3     ✓ stringr 1.4.0
✓ readr   2.0.1     ✓ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(ggbeeswarm)
library(phytools)</code></pre>
<pre><code>Loading required package: ape</code></pre>
<pre><code>Loading required package: maps</code></pre>
<pre><code>
Attaching package: &#39;maps&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    map</code></pre>
<pre class="r"><code>library(patchwork)
library(here)</code></pre>
<pre><code>here() starts at /Users/etytel01/Documents/Vertebrae/Code</code></pre>
<pre class="r"><code>library(ggtree)</code></pre>
<pre><code>ggtree v3.0.2  For help: https://yulab-smu.top/treedata-book/

If you use ggtree in published research, please cite the most appropriate paper(s):

1. Guangchuang Yu. Using ggtree to visualize data on tree-like structures. Current Protocols in Bioinformatics, 2020, 69:e96. doi:10.1002/cpbi.96
2. Guangchuang Yu, Tommy Tsan-Yuk Lam, Huachen Zhu, Yi Guan. Two methods for mapping and visualizing associated data on phylogeny using ggtree. Molecular Biology and Evolution 2018, 35(12):3041-3043. doi:10.1093/molbev/msy194
3. Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam. ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data. Methods in Ecology and Evolution 2017, 8(1):28-36. doi:10.1111/2041-210X.12628</code></pre>
<pre><code>
Attaching package: &#39;ggtree&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ape&#39;:

    rotate</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    expand</code></pre>
<pre class="r"><code>library(plotly)</code></pre>
<pre><code>
Attaching package: &#39;plotly&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    last_plot</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
<pre><code>The following object is masked from &#39;package:graphics&#39;:

    layout</code></pre>
<div id="figure-1" class="section level2">
<h2>Figure 1</h2>
<p>For this figure, we need to identify three species from the three habitat classes that have clearly different vertebrae.</p>
<pre class="r"><code>vertdata &lt;- read_csv(here(&#39;output/vertdata_summary_lm_species.csv&#39;))</code></pre>
<pre><code>Rows: 77 Columns: 99</code></pre>
<pre><code>── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr  (7): Species, Habitat, Water_Type, MatchSpecies, MatchGenus, FullName, ...
dbl (92): fineness, CBL_med, CBL_max, CBL_mn, d_med, d_max, d_mn, alphaAnt_m...</code></pre>
<pre><code>
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>pairs &lt;- read_csv(here(&#39;output/vertdata_pairs.csv&#39;))</code></pre>
<pre><code>Rows: 46 Columns: 104</code></pre>
<pre><code>── Column specification ────────────────────────────────────────────────────────
Delimiter: &quot;,&quot;
chr  (8): label, Species, Habitat, Water_Type, MatchSpecies, MatchGenus, Fam...
dbl (95): parent, node, branch.length, fineness, CBL_med, CBL_max, CBL_mn, d...
lgl  (1): diffhab</code></pre>
<pre><code>
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre class="r"><code>plot_ly(data = vertdata, type = &quot;scatter&quot;, mode = &quot;markers&quot;) %&gt;%
  add_trace(x = ~Habitat, y = ~d_med, type = &quot;box&quot;,
            text = ~Species, hoverinfo = &quot;text&quot;,
            boxpoints = &quot;all&quot;, jitter = 0.2)</code></pre>
<pre><code>Warning: Can&#39;t display both discrete &amp; non-discrete data on same axis</code></pre>
<pre><code>Warning: &#39;box&#39; objects don&#39;t have these attributes: &#39;mode&#39;
Valid attributes include:
&#39;type&#39;, &#39;visible&#39;, &#39;showlegend&#39;, &#39;legendgroup&#39;, &#39;opacity&#39;, &#39;uid&#39;, &#39;ids&#39;, &#39;customdata&#39;, &#39;meta&#39;, &#39;selectedpoints&#39;, &#39;hoverinfo&#39;, &#39;hoverlabel&#39;, &#39;stream&#39;, &#39;transforms&#39;, &#39;uirevision&#39;, &#39;y&#39;, &#39;x&#39;, &#39;x0&#39;, &#39;y0&#39;, &#39;dx&#39;, &#39;dy&#39;, &#39;xperiod&#39;, &#39;yperiod&#39;, &#39;xperiod0&#39;, &#39;yperiod0&#39;, &#39;xperiodalignment&#39;, &#39;yperiodalignment&#39;, &#39;name&#39;, &#39;q1&#39;, &#39;median&#39;, &#39;q3&#39;, &#39;lowerfence&#39;, &#39;upperfence&#39;, &#39;notched&#39;, &#39;notchwidth&#39;, &#39;notchspan&#39;, &#39;boxpoints&#39;, &#39;jitter&#39;, &#39;pointpos&#39;, &#39;boxmean&#39;, &#39;mean&#39;, &#39;sd&#39;, &#39;orientation&#39;, &#39;quartilemethod&#39;, &#39;width&#39;, &#39;marker&#39;, &#39;line&#39;, &#39;fillcolor&#39;, &#39;whiskerwidth&#39;, &#39;offsetgroup&#39;, &#39;alignmentgroup&#39;, &#39;selected&#39;, &#39;unselected&#39;, &#39;text&#39;, &#39;hovertext&#39;, &#39;hovertemplate&#39;, &#39;hoveron&#39;, &#39;xcalendar&#39;, &#39;ycalendar&#39;, &#39;xaxis&#39;, &#39;yaxis&#39;, &#39;idssrc&#39;, &#39;customdatasrc&#39;, &#39;metasrc&#39;, &#39;hoverinfosrc&#39;, &#39;ysrc&#39;, &#39;xsrc&#39;, &#39;q1src&#39;, &#39;mediansrc&#39;, &#39;q3src&#39;, &#39;lowerfencesrc&#39;, &#39;upperfencesrc&#39;, &#39;notchspansrc&#39;, &#39;meansrc&#39;, &#39;sdsrc&#39;, &#39;textsrc&#39;, &#39;hovertextsrc&#39;, &#39;hovertemplatesrc&#39;, &#39;key&#39;, &#39;set&#39;, &#39;frame&#39;, &#39;transforms&#39;, &#39;_isNestedKey&#39;, &#39;_isSimpleKey&#39;, &#39;_isGraticule&#39;, &#39;_bbox&#39;</code></pre>
<div id="htmlwidget-958f1bde87846b696546" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-958f1bde87846b696546">{"x":{"visdat":{"c6db18f18048":["function () ","plotlyVisDat"]},"cur_data":"c6db18f18048","attrs":{"c6db18f18048":{"mode":"markers","alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"scatter"},"c6db18f18048.1":{"mode":"markers","alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"box","x":{},"y":{},"text":{},"hoverinfo":"text","boxpoints":"all","jitter":0.2,"inherit":true}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"xaxis":{"domain":[0,1],"automargin":true,"title":"Habitat"},"yaxis":{"domain":[0,1],"automargin":true,"title":"d_med"},"hovermode":"closest","showlegend":true},"source":"A","config":{"showSendToCloud":false},"data":[{"mode":"markers","type":"scatter","marker":{"color":"rgba(31,119,180,1)","line":{"color":"rgba(31,119,180,1)"}},"error_y":{"color":"rgba(31,119,180,1)"},"error_x":{"color":"rgba(31,119,180,1)"},"line":{"color":"rgba(31,119,180,1)"},"xaxis":"x","yaxis":"y","frame":null},{"fillcolor":"rgba(255,127,14,0.5)","mode":"markers","type":"box","x":["pelagic","demersal","pelagic","demersal","benthic","pelagic","pelagic","pelagic","benthic","demersal","pelagic","benthic","demersal","pelagic","benthic","demersal","demersal","pelagic","pelagic","pelagic","benthic","demersal","pelagic","demersal","demersal","benthic","pelagic","demersal","demersal","demersal","benthic","pelagic","pelagic","demersal","demersal","demersal","demersal","demersal","demersal","benthic","benthic","pelagic","demersal","demersal","pelagic","benthic","demersal","demersal","pelagic","demersal","demersal","demersal","demersal","benthic","benthic","demersal","benthic","demersal","demersal","demersal","benthic","demersal","demersal","demersal","demersal","benthic","benthic","demersal","benthic","benthic","pelagic","demersal","pelagic","pelagic","demersal","demersal","pelagic"],"y":[0.00085,0.0017,0.00334999999999999,0.0056,0.0023,0.0008,0.0008,0.0008,0.00105,0.00145,0.00085,0.0014,0.0024,0.0004,0.0023,0.0027,0.0027,0.0001,0.00159999999999999,0.002,0.00185,0.00219999999999999,0.002396817,0.00124999999999999,0.00109999999999999,0.0011,0.0038,0.0007,0.0009,0.00124999999999999,0.0003,0.0005,0.00015,0.00505,0.0007,0.0012,0.0021,0.00675,0.00155,0.0022,0.00319999999999999,0.0029,0.00045,0.00185,0.0006,0.0026,0.0018,0.00115,0.00174999999999999,0.00045,0.0034,0.0062,0.00159999999999999,0.0019,0.001,0.00475,0.00285,0.0007,0.0003,0.001,0.0015,0.00055,0.00205,0.0022,0.0007,0.00165,0.00275,0.0023,0.0013,0.0018,0.0025,0.00174999999999999,0.001,0.0034,0.0029,0.0023,0.00075],"text":["Abramis_brama","Alectis_ciliaris","Alosa_pseudoharengus","Amia_calva","Ammodytes_personatus","Anodontostoma_chacunda","Anoplogaster_cornuta","Aphareus_furca","Apodichthys_flavidus","Apteronotus_cuchillejo","Aulorhynchus_flavidus","Barbichthys_laevis","Catostomus_catostomus","Centropomus_undecimalis","Cephalopholis_argus","Chaetostoma_lineopunctatum","Chanos_chanos","Cheilopogon_melanurus","Chiasmodon_niger","Coryphaena_hippurus","Coryphaenoides_filifer","Ctenochaetus_strigosus","Cymatogaster_aggregata","Cyprinus_carpio","Diplectrum_formosum","Discherodontus_ashmeadi","Engraulis_mordax","Equetus_lanceolatus","Fundulus_heteroclitus","Gadopsis_marmoratus","Halieutichthys_aculeatus","Hemitaurichthys_polylepis","Hirundichthys_rondeletii","Hypomesus_pretiosus","Hypsypops_rubicundus","Labroides_dimidiatus","Lagodon_rhomboides","Lepidogalaxias_salamandroides","Lepomis_macrochirus","Lumpenus_sagitta","Malacocottus_zonurus","Malacosteus_niger","Menidia_beryllina","Microgadus_proximus","Mugil_cephalus","Myoxocephalus_polyacanthocephalus","Myripristis_jacobus","Nannaethiops_unitaeniatus","Nealotus_tripes","Nectamia_fusca","Novumbra_hubbsi","Oncorhynchus_gorbuscha","Ophidion_scrippsae","Ophiodon_elongatus","Opsanus_pardus","Osteoglossum_bicirrhosum","Parophrys_vetulus","Percopsis_omiscomaycus","Phalacronotus_bleekeri","Pholis_laeta","Phytichthys_chirus","Pimelodella_cristata","Poecilia_reticulata","Polymixia_lowei","Polypterus_senegalus","Remora_remora","Risor_ruber","Roeboides_affinis","Ronquilus_jordani","Salarias_fasciatus","Scomber_japonicus","Serrasalmus_rhombeus","Sphyraena_sphyraena","Stenobrachius_leucopsarus","Stephanolepis_hispidus","Symphysodon_discus","Trachurus_trachurus"],"hoverinfo":["text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text"],"boxpoints":"all","jitter":0.2,"marker":{"color":"rgba(255,127,14,1)","line":{"color":"rgba(255,127,14,1)"}},"line":{"color":"rgba(255,127,14,1)"},"xaxis":"x","yaxis":"y","frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="a-tibble-14-100" class="section level1">
<h1>A tibble: 14 × 100</h1>
<p>Species Habitat Water_Type MatchSpecies MatchGenus fineness CBL_med CBL_max <chr> <chr> <chr> <chr> <chr> <dbl> <dbl> <dbl> 1 Coryphae… benthic marine armatus Coryphaen… 7.76 0.0250 0.0299 2 Ophiodon… benthic marine decagrammus Hexagramm… 8.86 0.0139 0.0141 3 Remora_r… benthic marine osteochir Remora 6.84 0.0286 0.0304 4 Salarias… benthic marine fasciatus Salarias 5.77 0.0241 0.0278 5 Alectis_… demers… marine ciliaris Alectis 8.75 0.0346 0.0357 6 Microgad… demers… marine proximus Microgadus 8.03 0.0169 0.0177 7 Myripris… demers… marine murdjan Myriprist… 4.31 0.0294 0.0333 8 Ophidion… demers… marine holbrookii Ophidion 11.6 0.0137 0.0163 9 Serrasal… demers… freshwater rhombeus Serrasalm… 6.41 0.0194 0.0236 10 Abramis_… pelagic freshwater alburnus Alburnus 8.95 0.0166 0.0177 11 Anodonto… pelagic freshwater cepedianum Dorosoma 4.66 0.0228 0.0238 12 Anoploga… pelagic marine cornuta Anoplogas… 5.04 0.0287 0.0329 13 Aphareus… pelagic marine furca Aphareus 5.01 0.0312 0.0334 14 Aulorhyn… pelagic marine flavidus Aulorhync… 13.1 0.0170 0.0210 # … with 92 more variables: CBL_mn <dbl>, d_med <dbl>, d_max <dbl>, d_mn <dbl>, # alphaAnt_med <dbl>, alphaAnt_max <dbl>, alphaAnt_mn <dbl>, # alphaPos_med <dbl>, alphaPos_max <dbl>, alphaPos_mn <dbl>, DAnt_med <dbl>, # DAnt_max <dbl>, DAnt_mn <dbl>, DPos_med <dbl>, DPos_max <dbl>, # DPos_mn <dbl>, dBW_med <dbl>, dBW_max <dbl>, dBW_mn <dbl>, # DAntBW_med <dbl>, DAntBW_max <dbl>, DAntBW_mn <dbl>, DPosBW_med <dbl>, # DPosBW_max <dbl>, DPosBW_mn <dbl>, d_normCBL_med <dbl>, …</p>
<p>Choose example species close to the median for their group:</p>
<ul>
<li>benthic: Barbichthys laevis (Sucker barb) or Myoxocephalus polyacanthocephalus (Sculpin)</li>
<li>demersal: Poecilia reticulata (Guppy)</li>
<li>pelagic: Sphyraena sphyraena (Barracuda)</li>
</ul>
<p>We’ll use the sculpin as an example benthic species, because we have good histology data for it.</p>
<pre class="r"><code>examplespecies &lt;- list(&quot;Myoxocephalus_polyacanthocephalus&quot;,
                    &quot;Poecilia_reticulata&quot;,
                    &quot;Sphyraena_sphyraena&quot;)</code></pre>
<pre class="r"><code>verttree &lt;- readRDS(here(&#39;output/vert_tree.rds&#39;))</code></pre>
<pre class="r"><code>vertdata %&gt;%
  filter(Species %in% examplespecies)</code></pre>
<pre><code># A tibble: 3 × 99
  Species    Habitat Water_Type MatchSpecies MatchGenus fineness CBL_med CBL_max
  &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt;      &lt;chr&gt;        &lt;chr&gt;         &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
1 Myoxoceph… benthic marine     polyacantho… Myoxoceph…     2.72  0.0191  0.0231
2 Poecilia_… demers… freshwater latipinna    Poecilia       5.94  0.0262  0.0301
3 Sphyraena… pelagic marine     sphyraena    Sphyraena     11.5   0.0321  0.0362
# … with 91 more variables: CBL_mn &lt;dbl&gt;, d_med &lt;dbl&gt;, d_max &lt;dbl&gt;, d_mn &lt;dbl&gt;,
#   alphaAnt_med &lt;dbl&gt;, alphaAnt_max &lt;dbl&gt;, alphaAnt_mn &lt;dbl&gt;,
#   alphaPos_med &lt;dbl&gt;, alphaPos_max &lt;dbl&gt;, alphaPos_mn &lt;dbl&gt;, DAnt_med &lt;dbl&gt;,
#   DAnt_max &lt;dbl&gt;, DAnt_mn &lt;dbl&gt;, DPos_med &lt;dbl&gt;, DPos_max &lt;dbl&gt;,
#   DPos_mn &lt;dbl&gt;, dBW_med &lt;dbl&gt;, dBW_max &lt;dbl&gt;, dBW_mn &lt;dbl&gt;,
#   DAntBW_med &lt;dbl&gt;, DAntBW_max &lt;dbl&gt;, DAntBW_mn &lt;dbl&gt;, DPosBW_med &lt;dbl&gt;,
#   DPosBW_max &lt;dbl&gt;, DPosBW_mn &lt;dbl&gt;, d_normCBL_med &lt;dbl&gt;, …</code></pre>
<pre class="r"><code>vertdata &lt;-
  vertdata %&gt;%
  mutate(WaterTypeShort = str_sub(Water_Type, start = 1, end = 1))</code></pre>
<pre class="r"><code>highlightpairs &lt;-
  pairs %&gt;%
  distinct(parent) %&gt;%
  pull(parent)</code></pre>
<pre class="r"><code>left_join(as_tibble(verttree), vertdata, by = c(&quot;label&quot; = &quot;FullName&quot;)) %&gt;%
  mutate(label = str_replace(Species, &quot;_&quot;, &quot; &quot;)) %&gt;%
  tidytree::as.treedata() %&gt;%
  ggtree() + # layout = &quot;circular&quot;, open.angle = 120) + 
  scale_y_reverse() +
  geom_tiplab(aes(color = Habitat), size=1.5, offset = 5) +
  geom_tippoint(aes(shape = Water_Type)) +
  geom_text2(aes(label=Species, subset=Species %in% examplespecies),
             hjust = 0, vjust = 0) +
  geom_hilight(mapping=aes(subset = node %in% highlightpairs)) +
  scale_shape_manual(values = c(3, 23, 24)) +
  scale_color_manual(values = c(benthic=&quot;chocolate4&quot;, demersal = &quot;gold&quot;, pelagic = &quot;deepskyblue2&quot;)) +
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<pre><code>Scale for &#39;y&#39; is already present. Adding another scale for &#39;y&#39;, which will
replace the existing scale.</code></pre>
<p><img src="figure/20-plot_phylogeny.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>  #geom_label2(aes(label=&#39;P&#39;, subset = ispair))</code></pre>
<pre class="r"><code>ggsave(here(&#39;output/plot_example_data_figure.pdf&#39;), width=3.5, height=6, units=&quot;in&quot;)</code></pre>
<pre class="r"><code>vertdata %&gt;%
  group_by(Habitat) %&gt;%
  summarize(n = n(), frac = n() / nrow(vertdata))</code></pre>
<pre><code># A tibble: 3 × 3
  Habitat      n  frac
  &lt;chr&gt;    &lt;int&gt; &lt;dbl&gt;
1 benthic     18 0.234
2 demersal    38 0.494
3 pelagic     21 0.273</code></pre>
<pre class="r"><code>vertdata %&gt;%
  group_by(Water_Type) %&gt;%
  summarize(n = n(), frac = n() / nrow(vertdata))</code></pre>
<pre><code># A tibble: 3 × 3
  Water_Type     n   frac
  &lt;chr&gt;      &lt;int&gt;  &lt;dbl&gt;
1 anadromous     1 0.0130
2 freshwater    26 0.338 
3 marine        50 0.649 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] plotly_4.9.4.1   ggtree_3.0.2     here_1.0.1       patchwork_1.1.1 
 [5] phytools_0.7-80  maps_3.3.0       ape_5.5          ggbeeswarm_0.6.0
 [9] forcats_0.5.1    stringr_1.4.0    dplyr_1.0.7      purrr_0.3.4     
[13] readr_2.0.1      tidyr_1.1.3      tibble_3.1.4     ggplot2_3.3.5   
[17] tidyverse_1.3.1 

loaded via a namespace (and not attached):
  [1] colorspace_2.0-2        ellipsis_0.3.2          rprojroot_2.0.2        
  [4] fs_1.5.0                aplot_0.1.0             rstudioapi_0.13        
  [7] farver_2.1.0            bit64_4.0.5             fansi_0.5.0            
 [10] lubridate_1.7.10        xml2_1.3.2              codetools_0.2-18       
 [13] mnormt_2.0.2            knitr_1.34              jsonlite_1.7.2         
 [16] workflowr_1.7.0         broom_0.7.9             dbplyr_2.1.1           
 [19] BiocManager_1.30.16     compiler_4.1.2          httr_1.4.2             
 [22] rvcheck_0.1.8           backports_1.2.1         assertthat_0.2.1       
 [25] Matrix_1.3-4            fastmap_1.1.0           lazyeval_0.2.2         
 [28] cli_3.0.1               later_1.3.0             htmltools_0.5.2        
 [31] tools_4.1.2             igraph_1.2.6            coda_0.19-4            
 [34] gtable_0.3.0            glue_1.4.2              clusterGeneration_1.3.7
 [37] fastmatch_1.1-3         Rcpp_1.0.7              cellranger_1.1.0       
 [40] vctrs_0.3.8             nlme_3.1-153            crosstalk_1.1.1        
 [43] xfun_0.25               rvest_1.0.1             lifecycle_1.0.0        
 [46] renv_0.14.0             phangorn_2.7.1          MASS_7.3-54            
 [49] scales_1.1.1            vroom_1.5.4             hms_1.1.0              
 [52] promises_1.2.0.1        parallel_4.1.2          expm_0.999-6           
 [55] yaml_2.2.1              yulab.utils_0.0.2       stringi_1.7.4          
 [58] highr_0.9               plotrix_3.8-2           tidytree_0.3.5         
 [61] rlang_0.4.11            pkgconfig_2.0.3         evaluate_0.14          
 [64] lattice_0.20-45         labeling_0.4.2          treeio_1.16.1          
 [67] htmlwidgets_1.5.4       bit_4.0.4               tidyselect_1.1.1       
 [70] magrittr_2.0.1          R6_2.5.1                generics_0.1.0         
 [73] combinat_0.0-8          DBI_1.1.1               pillar_1.6.2           
 [76] haven_2.4.3             whisker_0.4             withr_2.4.2            
 [79] scatterplot3d_0.3-41    modelr_0.1.8            crayon_1.4.1           
 [82] utf8_1.2.2              tmvnsim_1.0-2           tzdb_0.1.2             
 [85] rmarkdown_2.10          grid_4.1.2              readxl_1.3.1           
 [88] data.table_1.14.0       git2r_0.29.0            reprex_2.0.1           
 [91] digest_0.6.27           httpuv_1.6.4            numDeriv_2016.8-1.1    
 [94] gridGraphics_0.5-1      munsell_0.5.0           beeswarm_0.4.0         
 [97] viridisLite_0.4.0       ggplotify_0.1.0         vipor_0.4.5            
[100] quadprog_1.5-8         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
